mmv1/products/compute/NetworkEndpoints.yaml

```diff
@@ -0,0 +1,134 @@
+read_verb: :POST
+self_link: projects/{{project}}/zones/{{zone}}/networkEndpointGroups/{{network_endpoint_group}}/listNetworkEndpoints
+identity:
+  - networkEndpointGroup
+collection_url_key: 'items'
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/load-balancing/docs/negs/'
+  api: 'https://cloud.google.com/compute/docs/reference/rest/beta/networkEndpointGroups'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    kind: 'compute#operation'
+    path: 'name'
+    base_url: 'projects/{{project}}/zones/{{zone}}/operations/{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'targetLink'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'status'
+    complete: 'DONE'
+    allowed:
+      - 'PENDING'
+      - 'RUNNING'
+      - 'DONE'
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error/errors'
+    message: 'message'
+id_format: '{{project}}/{{zone}}/{{network_endpoint_group}}/endpoints'
+mutex: networkEndpoint/{{project}}/{{zone}}/{{network_endpoint_group}}
+exclude_validator: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_endpoints'
+    primary_resource_id:
+      'default-endpoints'
+      # Fine-grained resource need different autogenerated tests, as
+      # we need to check destroy during a test step where the parent resource
+      # still exists, rather than during CheckDestroy (when read returns
+      # nothing because the parent resource has then also been destroyed)
+    skip_test: true
+    vars:
+      neg_name: 'my-lb-neg'
+      instance_name: 'endpoint-instance'
+      network_name: 'neg-network'
+      subnetwork_name: 'neg-subnetwork'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  pre_delete: templates/terraform/pre_delete/compute_network_endpoints.go.erb
+  custom_import: templates/terraform/custom_import/compute_network_endpoints.go.erb
+  decoder: templates/terraform/decoders/network_endpoints.go.erb
+  encoder: templates/terraform/encoders/compute_network_endpoints.go.erb
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'zone'
+    resource: 'Zone'
+    imports: 'name'
+    description: |
+      Zone where the containing network endpoint group is located.
+    required: false
+    url_param_only: true
+    default_from_api: true
+    ignore_read: true
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'networkEndpointGroup'
+    resource: 'NetworkEndpointGroup'
+    imports: 'name'
+    description: |
+      The network endpoint group this endpoint is part of.
+    required: true
+    url_param_only: true
+    ignore_read: true
+    diff_suppress_func: tpgresource.CompareResourceNames
+properties:
+  - !ruby/object:Api::Type::Array
+    name: 'endpoints'
+    api_name: 'networkEndpoints'
+    is_set: true
```

Comment:
in general we try to avoid sets - they are less useful to users. For example, a common Terraform use case is referencing an item by its index in a list, which doesn't work for sets. The one issue is that the list returned by the API needs to have a stable sort, and the list in the config needs to match that sort during diffing.  I believe a diff suppress func would be sufficient: https://googlecloudplatform.github.io/magic-modules/how-to/add-mmv1-resource/#diffsuppressfunc

Reply:
Sounds good, I also found `unordered_list`. Does that have the same semantics?

Reply:
`unordered_list` apparently ["Indicates that this is an Array that should have Set diff semantics."](https://github.com/GoogleCloudPlatform/magic-modules/blob/49d180598e0ad80d6b3a05f050ea4aaefa4b0bdd/mmv1/api/type.rb#L112) It's currently only used on one resource in the entire provider, so I'd be wary of it.

After more discussion - it sounds like we might prefer a TypeSet here actually, especially if the API does not allow duplicates - does it? Still an ongoing discussion but TypeSet is safe if you want to just stick with that for now.

Sorry for the back and forth on this. Lists / sets are just generally tricky to manage well in Terraform.

Reply:
No worries, I can imagine it's complicated to pick between the two.

The API does not allow duplicates, but I've been struggling to get Terraform to even provide duplicates in the first place. I changed it back to TypeSet since it matches the API better. I guess users can always cast it to a list if they need to.